### PR TITLE
My location button

### PR DESCRIPTION
**Pull request by TheImpressiveDonut**

### CODE REVIEW

File: app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java [hunk 1]
```diff
@@ -142,6 +168,18 @@ public void testAddEventPutsMarkerOnMap() {
         onView(withId(R.id.fragment_map_event_name)).check(matches(withText(TEST_EVENT_1.getName())));
     }
 
+    @Test
+    @Ignore("Cirrus reject")
+    public void testMyLocationIsAppearing() {
```
Comment (Ef55) [inline app/src/androidTest/java/com/github/onedirection/map/MapFragmentTest.java:173]: Happy to know that I'm not the only one with this kind of problems.

**Review by Ef55 — CHANGES_REQUESTED** [review]
Only one issue: there are some Linter's warnings.
(Which I could not highlight for some reason).

**Review by Ef55 — APPROVED** [review]

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 2]
```diff
@@ -38,14 +46,26 @@
...
+
     public static final String SYMBOL_ID = "MARKER_MAP";
+    public static final String MY_LOCATION_ID = "MY_LOCATION_MAP";
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater,
                              @Nullable ViewGroup container,
                              @Nullable Bundle savedInstanceState) {
+        deviceLocationProvider = (DeviceLocationProvider) requireActivity();
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:61]: I don't like the unconditionnal cast, but i don't know how you could have made it better

**Review by Sosol02 — APPROVED** [review]

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java (hunk 2 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:61]: We should be able to do better after PR #109

**Pull request merged**